daemon/graphdriver/overlay2/check.go

```diff
@@ -75,5 +81,23 @@ func hasOpaqueCopyUpBug(d string) error {
 		return errors.New("opaque flag erroneously copied up, consider update to kernel 4.8 or later to fix")
 	}
 
+	// rename "d1" to "d2"
+	if err := os.Rename(filepath.Join(td, "merged", "d1"), filepath.Join(td, "merged", "d2")); err != nil {
+		// if rename fail with syscall.EXDEV, the kernel doesn't enable CONFIG_OVERLAY_FS_REDIRECT_DIR
+		if strings.Contains(err.Error(), syscall.EXDEV.Error()) {
```

Comment:
Any reason we can't check the error number directly? This can be downcast to an `(*os.LinkError)` and you can check against `(*os.LinkError).Err`. See https://golang.org/src/os/file_unix.go#L19 to see how the error is packed.